File: 1979-find-greatest-common-divisor-of-array/1979-find-greatest-common-divisor-of-array.java
class Solution {
    public int findGCD(int[] nums) {
        int Max = nums[0];
        int Min = nums[0];
        for(int i=0;i<nums.length;i++){
            Max = Math.max(Max,nums[i]);
            Min = Math.min(Min,nums[i]);
        }
        return gcd(Min,Max);
    }
    
    public static int gcd(int a, int b) {
        while (b != 0) {
            int temp = b;
            b = a % b;
            a = temp;
        }
        return a;
    }
}

File: 1979-find-greatest-common-divisor-of-array/1979-find-greatest-common-divisor-of-array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findGCD = function(nums) {
    nums.sort((a,b) => a-b);
    let min = nums[0], max = nums[nums.length-1];
    return gcd(min,max);
    
    function gcd(a,b){
        while(b!=0){
            let temp = b;
            b=a%b;
            a = temp;
        }
        return a;
    }
};